app/models/manageiq/providers/ibm_cloud/object_storage/storage_manager.rb

```diff
@@ -0,0 +1,211 @@
+                        :id         => "authentications.bearer.userid",
+                        :label      => _("Access Key"),
+                        :isRequired => true,
+                        :validate   => [{:type => "required"}],
+                      },
+                      {
+                        :component  => "text-field",
+                        :name       => "authentications.bearer.password",
+                        :id         => "authentications.bearer.password",
+                        :label      => _("Secret Key"),
+                        :isRequired => true,
+                        :validate   => [{:type => "required"}],
+                        :type       => "password",
+                      }
+                    ]
+                  }
+                ]
+              }
+            ]
+          ]
+        }
+      ]
+    }.freeze
+  end
+
+  def self.hostname_required?
+    false
+  end
+
+  def self.supported_for_create?
+    true
+  end
+
+  def self.ems_type
+    @ems_type ||= "ibm_cloud_object_storage".freeze
+  end
+
+  def self.description
+    @description ||= "IBM Cloud Object Storage".freeze
+  end
+
+  def image_name
+    "ibm"
+  end
+
+  def cos_creds
+    iam     = authentications.detect { |e| e.authtype == 'default' }
+    bearer  = authentications.detect { |e| e.authtype == 'bearer' }
+    endp    = endpoints.first
+
+    guid   = uid_ems
+    region = provider_region
+    endpoint = endp.url
+    apikey   = iam.auth_key
+    access_key = bearer.userid
+    secret_encr = bearer.password
+    secret_key  = ManageIQ::Password.try_decrypt(secret_encr)
+
+    return guid, apikey, region, endpoint, access_key, secret_key
+  end
+
+  def connect(_args = {})
+    _, _, region, endpoint, access_key, secret_key = cos_creds
+    self.class.raw_connect(region, endpoint, access_key, secret_key)
+  end
+
+  def verify_credentials(_args = {})
+    guid, apikey, region, endpoint, access_key, secret_key = cos_creds
+    self.class.verify_bearer(region, endpoint, access_key, secret_key)
+    self.class.verify_iam(guid, apikey)
+
+    true
+  end
+
+  def self.verify_credentials(args)
+    guid     = args["uid_ems"]
+    auth_key = args.dig("authentications", "default", "auth_key")
+    apikey   = ManageIQ::Password.try_decrypt(auth_key)
+    verify_iam(guid, apikey)
+
+    region = args["provider_region"]
+    endpoint = args.dig("endpoints", "default", "url")
+    access_key = args.dig("authentications", "bearer", "userid")
+    secret_encr = args.dig("authentications", "bearer", "password")
+    secret_key  = ManageIQ::Password.try_decrypt(secret_encr)
+    verify_bearer(region, endpoint, access_key, secret_key)
+
+    true
+  end
+
+  def self.verify_bearer(region, endpoint, access_key, secret_key)
+    begin
+      raw_connect(region, endpoint, access_key, secret_key).list_buckets({}, :params => {:max_keys => 1})
+    rescue Aws::Errors::ServiceError, Seahorse::Client::NetworkingError => err
```

Comment:
Are we really expecting AWS errors here?

Reply:
Yes, IBM uses an S3 compatible API so @iv1111 is using the `Aws::S3` gem: https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/159/files#diff-ffe546bf7942848309735f1742f214e4af3c09c6bbacdf3ef0d1277981902b02R197-R209